nautobot/extras/nautobot_app/__init__.py

```diff
@@ -0,0 +1,136 @@
+from django.apps import AppConfig
+from operator import getitem
+from collections import OrderedDict
+
+from nautobot.extras.plugins.utils import import_object
+from nautobot.extras.registry import registry
+from nautobot.utilities.choices import ButtonColorChoices
+
+
+class NautobotConfig(AppConfig):
+    menu_tabs = "navigation.menu_tabs"
+
+    def ready(self):
+        if self.__module__.endswith(".apps"):
+            self.__module__ = self.__module__[:-5]
+        menu_items = import_object(f"{self.__module__}.{self.menu_tabs}")
+        if menu_items is not None:
+            register_menu_items(menu_items)
+
+
+def register_menu_items(class_list):
+    for nav_tab in class_list:
+        registry["nav_menu"]["tabs"][nav_tab.name] = {
```

Comment:
Need to protect against the case where a registry entry already exists for this `nav_tab.name` to avoid overwriting an existing menu with a new menu.

More generally, how *does* one add items to an existing menu without overwriting it? I don't see how the current design facilitates that.

Reply:
```
"groups": registry["nav_menu"]["tabs"][nav_tab.name]["groups"]
            if registry["nav_menu"]["tabs"].get(nav_tab.name)
            else {},
            "permissions": [],
```
The reason I do this is so a lower down plugin could affect the weight of a tab that already exists.  The code above adds in the groups to the overwritten tab.